### DISCUSSION TITLE
Problemas con el Maven-Java

### DISCUSSION THREAD

**Discussion by lgo1980** [dds-jv/foro#411] [Consultas de herramientas] [answered]
Buenas! Tengo problemas con el maven en mi compu, les paso a especificar, instale en mis 2 compus (Java, intelij, maven, etc) en una tengo zorin (es base de ubuntu) y la otra es win11(Pero uso una maquina virtual con virtualbox que tiene ubuntu 22.04), segui los pasos de este tutorial por si quieren ver -> https://www.youtube.com/watch?v=SfChLB1q3cE, en la maquina que tiene zorin (que es la de mi trabajo que no puedo usarla mucho) funciona bien, pero en mi compu con la mv con ubuntu, cada vez que arranco tanto en una terminal como en el intellij idea me pone este error cuando quiero correr cualquier comando de mvn (por ejemplo mvn test): 
![WhatsApp Image 2024-04-15 at 10 33 59](https://github.com/dds-jv/foro/assets/12699830/71941dc8-df1c-443a-9081-779b9a54318e)
y tengo que correr estos 2 comando para que me funcione:
chmod +x /etc/profile.d/maven.sh
source /etc/profile.d/maven.sh
La verdad que no se porque en una si y la otra no, lo unico que en mi compu habia tocado algo antes y puede ser que eso haya afectado, pero no se como arreglarlo.
No es tan grave , pero bueno si tienen alguna idea de como podria solucionarlo, buenisimo
Saludos
Leonardo Olmedo

**Comment by flbulgarelli** [comment]
¡Hola! Por lo que veo en tu captura de pantalla, el problema sería que `JAVA_HOME` apunta a un JRE (es decir, la máquina virtual estándar de Java), no un JDK (es decir, la máquina virtual más las herramientas de desarrollo como el compilador `javac`). 

Esto puede deberse a varios factores: 

1. ¿Puede que hayas instalado sólo el jre? Asegurate de que hayas instalado, por ejemplo `openjdk-17-jdk` en lugar de `openjdk-17-jre` (o similar)
2. ¿Puede que no hayas seteado la variable `JAVA_HOME` en tu bashrc? La misma debería apuntar a algo como por ejemplo `/usr/lib/jvm/java-1.17.0-openjdk-amd64/`

Por otro lado, hay algo que no termino de entender de tu consulta: entiendo que decís que no estás pudiendo hacerlo andar en la máquina virtual de Ubuntu en Windows, pero las capturas de pantalla parecen ser de una terminal nativa de Windows. ¿Qué máquina virtual estás usando? 

¡Saludos!

**Reply by lgo1980** [reply]
Hola! la captura la saque de internet porque no estaba con mi compu, es mas, la 3 linea que dice lo de JDK not a JRE no esta!, solo las 2 primeras! Creo que la variable JAVA_HOME en el video me la hizo poner en -> /etc/profile.d/maven.sh creo que en el lugar que decis no la tengo. (Va creo que la borre porque tambien me tiraba error ahi o crei que ese era el error)

**Reply by flbulgarelli** [reply]
OK. Cuando puedas subí entonces las capturas de tu propio sistema, así podemos darte una respuesta más precisa.  Es importante que nos digas el valor exacto de `JAVA_HOME`  (ejecutá `echo $JAVA_HOME`) y nos muestres la captura / el texto exacto de los mensajes de error que estás obteniendo. 

Saludos

**Comment by lgo1980** [comment]
Hola! Perdon recien estoy en mi compu, hice lo que me dijiste mas el error que te decia, saque captura:
![image](https://github.com/dds-jv/foro/assets/12699830/bb173ddc-c6e5-4a2d-8cfd-8c752c0ec4e8)
Capaz tengo que hacer lo del archivo de bash no se cuanto.
Saludos

**Reply by lgo1980** [reply, marked as answer]
Hola! Ya esta lo pude solucionar, era eso! estaba mal el archivo bashrc, ya esta perdon por las molestias

**Reply by flbulgarelli** [reply]
¡Genial! No hay problema. Un detalle para el futuro: tené en cuenta que en Linux las variables se referencian con el signo _pesos_, no con `%`. El uso del signo porcentaje es exclusivo de Windows.
